operatorcert/pyxis.py

```diff
@@ -23,3 +23,38 @@ def post_with_api_key(url: str, body: Dict[str, Any]) -> Dict[str, Any]:
         )
         raise
     return resp.json()
+
+
+def post_with_cert(
+    url: str, body: Dict[str, Any], cert_path: str, key_path: str
+) -> Dict[str, Any]:
+    if not os.path.exists(cert_path):
+        raise Exception(f"Pyxis cert at path {cert_path} is missing.")
+    if not os.path.exists(key_path):
+        raise Exception(f"Pyxis key at path {key_path} is missing.")
+
+    LOGGER.debug(f"POST Pyxis request: {url}")
+    resp = requests.post(url, json=body, cert=(cert_path, key_path))
+
+    try:
+        resp.raise_for_status()
+    except requests.HTTPError:
+        LOGGER.exception(
+            f"Pyxis POST query failed with {url} - {resp.status_code} - {resp.text}"
+        )
+        raise
+    return resp.json()
+
+
+def get_with_cert(url: str, cert_path: str, key_path: str) -> Dict[str, Any]:
+    if not os.path.exists(cert_path):
+        raise Exception(f"Pyxis cert at path {cert_path} is missing.")
+    if not os.path.exists(key_path):
+        raise Exception(f"Pyxis key at path {key_path} is missing.")
+
+    LOGGER.debug(f"GET Pyxis request: {url}")
+    resp = requests.get(url, cert=(cert_path, key_path))
+    # Not raising exception for error statuses, because GET request can be used to check
+    # if something exists. We don't want a 404 to cause failures.
+
+    return resp
```

Comment:
FYI, there seems to be a lot of impending overlap in #14. It might be worthwhile to align this with @Allda's change after it's merged.

Reply:
Yeah I noticed...I'd rather create another PR to fix it after though since Ales's change is also pending. This is ready if you wanna have another look.

Reply:
Sure, that's fine.